apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -87,24 +96,49 @@ func (api *UserManagerAPI) AddUser(ctx context.Context, args params.AddUsers) (p
 	}
 
 	for i, arg := range args.Users {
-		var user *state.User
-		var err error
+		// Get creator user
+		creator, err := api.userService.GetUserByName(ctx, api.apiUser.Id())
```

Comment:
I realise this decision probably predates this PR, but the paradigm here is inefficient and unnecessary. We are making a service call with a user name and are hydrating a full user entity and all we are using it for is to et the UUID to pass back to a subsequent service call to add a user.  A better, more efficient approach which also does not break encapsulation is to simply pass the username which we get off the wire to the service, and have the add user operation perform a look up to get the creator uuid as part of the create operation. This can even be done as a subselect in the insert sql.

The same comment applies to the user remove - just pass the name of the user to remove to the service.